### ISSUE TITLE
Dynamic groups from application.yml

### ISSUE THREAD

**Issue by YunaBraska** [springdoc/springdoc-openapi#344]
I would love to have a dynamically loaded group config. I failed myself using the BeanFactoryPostProcessor. For some reason it could not find the path to the swagger-json file.

**This is an example how it could look like:**
```
group-config:
    -
      name: "external"
      packages-to-scan: "com.example.app.controler.api"
      paths: "/api/**"
    -
      name: "internal"
      packages-to-scan: "com.example.app.controler.web"
      paths: "/capi/**"
```

**My approach:**
```
@Bean
    public static BeanFactoryPostProcessor beanFactoryPostProcessor(
            final Environment environment) {
        return beanFactory -> {
            final BindResult<SpringDocGroupConfig> result = Binder.get(environment)
                    .bind("springdoc", SpringDocGroupConfig.class);
            final SpringDocGroupConfig groupConfig = result.get();
            // Use the properties to post-process the bean factory as needed
            if (groupConfig != null && groupConfig.getGroupConfig() != null) {
                groupConfig.getGroupConfig().forEach(config -> {
                    final GroupedOpenApi.Builder builder = GroupedOpenApi.builder();
                    if (hasText(config.getName())) {
                        builder.setGroup(config.getName());
                    }
                    if (config.getPaths() != null) {
                        builder.pathsToMatch(config.getPaths().toArray(new String[0]));
                    }
                    if (config.getPackagesToScan() != null) {
                        builder.packagesToScan(config.getPackagesToScan().toArray(new String[0]));
                    }
                    final GroupedOpenApi groupedOpenApi = builder.build();
                    beanFactory.registerSingleton(groupedOpenApi.getGroup(), groupedOpenApi);
                });
            }
        };
    }
```

**Comment by bnasslahsen** [comment]
Hi @YunaBraska, 

This seems an interesting feature and you were too close to achieve it.
Here is a sample code that should help:

```java
@Bean
public BeanFactoryPostProcessor beanFactoryPostProcessor(Environment environment) {
	return beanFactory -> {
		final BindResult<SpringDocGroupConfig> result = Binder.get(environment)
				.bind("springdoc", SpringDocGroupConfig.class);
		SpringDocGroupConfig springDocGroupConfig =result.get();
		List<GroupedOpenApi> groupedOpenApis = springDocGroupConfig.getGroupsOpenAPIList().stream()
				.map( elt -> {
					GroupedOpenApi.Builder builder = GroupedOpenApi.builder();
					if(!CollectionUtils.isEmpty(elt.getPackagesToScan()))
						builder.packagesToScan(elt.getPackagesToScan().toArray(new String[0]));
					if(!CollectionUtils.isEmpty(elt.getPathsToMatch()))
						builder.pathsToMatch(elt.getPathsToMatch().toArray(new String[0]));
					return builder.setGroup(elt.getGroup()).build();
				})
				.collect(Collectors.toList());
		groupedOpenApis.forEach( elt -> beanFactory.registerSingleton(elt.getGroup(), elt));
		for (String beanName : beanFactory.getBeanNamesForType(OpenAPIBuilder.class)) {
			beanFactory.getBeanDefinition(beanName).setScope("prototype");
		}
		for (String beanName : beanFactory.getBeanNamesForType(OpenAPI.class)) {
			beanFactory.getBeanDefinition(beanName).setScope("prototype");
		}
	};
}

@Bean
public MultipleOpenApiResource multipleOpenApiResource(List<GroupedOpenApi> groupedOpenApis,ObjectFactory<OpenAPIBuilder> defaultOpenAPIBuilder, AbstractRequestBuilder requestBuilder,AbstractResponseBuilder responseBuilder, OperationBuilder operationParser,RequestMappingInfoHandlerMapping requestMappingHandlerMapping, Optional<ActuatorProvider> servletContextProvider) {
	return new MultipleOpenApiResource(groupedOpenApis,
			defaultOpenAPIBuilder, requestBuilder,
			responseBuilder, operationParser,
			requestMappingHandlerMapping, servletContextProvider);
}
```

**Comment by YunaBraska** [comment]
@bnasslahsen why did you close this interesting feature? At least I was close to archive this :D But I don't want to write this in every micro service. would be really nice if you could just implement that feature as you can see I can't do this without your awesome help.

**Comment by bnasslahsen** [comment]
Hi @YunaBraska, 

 My answer is for your particular description to help you in short term.
 This feature will be added certainly for a new release.

**Comment by bnasslahsen** [comment]
The dynamic groups support will be available with v1.3.1.

You will be able to define a group, using spring-boot application.yml.
Note that, for this usage, you don't have to declare the **GroupedOpenApi** Bean.

You need to declare the following properties, under the prefix **springdoc.group-configs**.

The list of properties under this prefix, are available here:
- https://springdoc.org/springdoc-properties.html#springdoc-openapi-core-properties
